zebra-chain/src/shutdown.rs

```diff
@@ -0,0 +1,26 @@
+//! Shutdown related code.
+//!
+//! A global flag indicates when the application is shutting down so actions can be taken
+//! at different parts of the codebase.
+
+use std::sync::atomic::{AtomicBool, Ordering};
+
+/// A flag to indicate if zebrad is shutting down.
+pub static IS_SHUTTING_DOWN: AtomicBool = AtomicBool::new(false);
```

Comment:
Two tweaks:
1. Let's just say `Zebra`, because we might also want to shut down `zebra-client`. And users of `zebra-network` might want to shut it down.
2. Let's be explicit about the initial value.

```suggestion
/// A flag to indicate if Zebra is shutting down.
///
/// Initialized to `false` at startup.
pub static IS_SHUTTING_DOWN: AtomicBool = AtomicBool::new(false);
```